src/odemis/acq/millmng.py

```diff
@@ -0,0 +1,504 @@
+                # Move the stage
+                target_pos = {"x": site.pos.value[0],
+                              "y": site.pos.value[1],
+                              "z": site.pos.value[2]}
+                # Todo check if it is wise to move in Z direction without affecting the OPTICAL-FIB alignment
+                logging.debug("For feature %s moving the stage to %s m", site.name.value, target_pos)
+                self._future.running_subf = self._stage.moveAbs(target_pos)
+                stage_time = self.estimate_stage_movement_time(site)
+                t = stage_time + 5  # adding extra margin
+                try:
+                    self._future.running_subf.result(t)  # or 2 min should be fast enough to move anywhere
+                except TimeoutError:
+                    logging.warning("Failed to move the stage for feature %s within %s s", site.name.value, t)
+                    self._future.running_subf.cancel()
+                    continue  # Try the next feature
+
+                # For each milling setting mill a rectangle for the given site
+                for setting_nb, milling_settings in enumerate(self._settings):
+                    # Tilt stage angle
+                    stage_rx = self._ion_beam_angle - milling_settings.beamAngle.value
+                    logging.debug("Tilting stage to %f °", math.degrees(stage_rx))
+                    self._future.running_subf = self._stage.moveAbs({"rx": stage_rx})
+
+                    try:
+                        t = 15  # s
+                        self._future.running_subf.result(timeout=t)
+                    except TimeoutError:
+                        logging.warning("Failed to tilt the stage for feature %s within %s s", site.name.value, t)
+                        self._future.running_subf.cancel()
+                        logging.debug("Cancelling the milling")
+                        self._future._task_state = CANCELLED
+                        raise CancelledError()
+
+                    # Configure current
+                    self._scanner.current.value = milling_settings.current.value
+
+                    # Configure HoV
+                    self._scanner.horizontalFoV.value = milling_settings.horizontalFoV.value
+
+                    # Initialize the drift corrector
+                    # max_pixels can change according to drift correction approach
+                    drift_est = AnchoredEstimator(self._scanner, self._detector,
+                                                  milling_settings.dcRoi.value,
+                                                  milling_settings.dcDwellTime.value,
+                                                  max_pixels=512 ** 2, follow_drift=False)
+                    # acquire an image at the given location (RoI)
+                    drift_est.acquire()
+                    da = drift_est.raw[-1]
+                    pixel_size = da.metadata[model.MD_PIXEL_SIZE]
```

Comment:
That's not a very specific name... especially at the next variable is called `pxs`! 
I'd suggest `pxs_drift` or `pxs_anchor`.